hplsql/src/main/java/org/apache/hive/hplsql/functions/Function.java

```diff
@@ -1,780 +1,30 @@
-    try {
-      while (rs.next()) {
-        if (rs.getString(1).startsWith("Detailed Partition Information")) {
-          Matcher m = Pattern.compile(".*, location:(.*?),.*").matcher(rs.getString(2));
-          if (m.find()) {
-            result = m.group(1);
-          }    
-        }
-      }
-    } catch (SQLException e) {}  
-    if (result != null) {
-      // Remove the host name
-      if (!hostname) {
-        Matcher m = Pattern.compile(".*://.*?(/.*)").matcher(result); 
-        if (m.find()) {
-          result = m.group(1);
-        }
-      }
-      evalString(result);
-    }    
-    else {
-      evalNull();
-    }
-    exec.closeQuery(query, exec.conf.defaultConnection);
-  }
-  
-  /**
-   * Evaluate the expression and push the value to the stack
-   */
-  void eval(ParserRuleContext ctx) {
-    exec.visit(ctx);
-  }
-
-  /**
-   * Evaluate the expression to the specified variable
-   */
-  void evalVar(Var var) {
-    exec.stackPush(var); 
-  }
-
-  /**
-   * Evaluate the expression to NULL
-   */
-  void evalNull() {
-    exec.stackPush(Var.Null); 
-  }
-  
-  /**
-   * Evaluate the expression to specified String value
-   */
-  void evalString(String string) {
-    exec.stackPush(new Var(string)); 
-  }
-  
-  void evalString(StringBuilder string) {
-    evalString(string.toString()); 
-  }
-
-  /**
-   * Evaluate the expression to specified Int value
-   */
-  void evalInt(Long i) {
-    exec.stackPush(new Var(i)); 
-  }
-  
-  void evalInt(int i) {
-    evalInt(Long.valueOf(i));
-  }
-  
-  /**
-   * Evaluate the expression to specified Date value
-   */
-  void evalDate(Date date) {
-    exec.stackPush(new Var(Var.Type.DATE, date)); 
-  }
-  
-  /**
-   * Evaluate the expression to NULL and close the query
-   */
-  void evalNullClose(Query query, String conn) {
-    exec.stackPush(Var.Null); 
-    exec.closeQuery(query, conn);
-    if(trace) {
-      query.printStackTrace();
-    }
-  }
-  
-  /**
-   * Evaluate the expression and pop value from the stack
-   */
-  Var evalPop(ParserRuleContext ctx) {
-    exec.visit(ctx);
-    return exec.stackPop();  
-  }
-  
-  Var evalPop(ParserRuleContext ctx, int value) {
-    if (ctx != null) {
-      return evalPop(ctx);
-    }
-    return new Var(Long.valueOf(value));
-  }
-  
-  /**
-   * Execute rules
-   */
-  Integer visit(ParserRuleContext ctx) {
-    return exec.visit(ctx);  
-  } 
- 
-  /**
-   * Execute children rules
-   */
-  Integer visitChildren(ParserRuleContext ctx) {
-    return exec.visitChildren(ctx);  
-  }  
-  
-  /**
-   * Trace information
-   */
-  public void trace(ParserRuleContext ctx, String message) {
-    if (trace) {
-      exec.trace(ctx, message);
-    }
-  }
-  
-  public void trace(String message) {
-    trace(null, message);
-  }
-  
-  public void info(ParserRuleContext ctx, String message) {
-    exec.info(ctx, message);
-  }
+public interface Function {
+  boolean exec(String name, HplsqlParser.Expr_func_paramsContext ctx);
```

Comment:
true: function was found and executed, false if it was not found. The return value was already used and expected by the existing code so I kept it. There are some issues with how the existing implementation works, for example calling an undefined function evaluates to null, instead of throwing an error. We might want to revisit these things later, but this wasn't the scope of this patch.